core/lib/core/observatory.ex

```diff
@@ -0,0 +1,54 @@
+defmodule Core.Observatory do
+  alias CoreWeb.Endpoint
+
+  def subscribe(signal, key \\ []) do
+    Endpoint.subscribe(topic_key(signal, key))
+  end
+
+  def dispatch(signal, key, data) do
+    Endpoint.broadcast(topic_key(signal, key), "observation", {signal, data})
+  end
+
+  def local_dispatch(signal, key, data) do
+    Endpoint.local_broadcast(topic_key(signal, key), "observation", {signal, data})
+  end
+
+  defp topic_key(signal, key) when is_atom(signal) and is_list(key) do
+    key_str =
+      key
+      |> Enum.map(&to_string/1)
+      |> Enum.join(":")
+
+    "signal:#{to_string(signal)}:#{key_str}"
+  end
+
+  def observe(socket, subscriptions \\ []) do
+    if Phoenix.LiveView.connected?(socket) do
+      for {signal, key} <- subscriptions do
+        __MODULE__.subscribe(signal, key)
+      end
+    end
+  end
+
+  defmacro __using__(_opts \\ []) do
+    quote do
+      @before_compile Core.Observatory
+      import unquote(__MODULE__), only: [observe: 2]
+
+      def handle_info(
+            %{topic: _topic, payload: {signal, message}},
+            socket
+          ) do
+        {:noreply, handle_observation(socket, signal, message)}
+      end
+    end
+  end
+
+  defmacro __before_compile__(_env) do
+    quote do
+      def handle_observation(signal, _message, _assigns) do
+        throw("No handler for observed signal: #{inspect(signal)}")
```

Comment:
Dit is tijdelijk neem ik aan? Op zich is dit een valide situatie, niet wenselijk maar wel mogelijk.

Reply:
Dit kan alleen gebeuren als je een `observe` doet op  een signal maar daar geen handler voor maakt. Dat lijkt me wel een fout (of de observe moet weg of je bent een handler vergeten). Wat voor use-case zie jij dat dit wel moet kunnen?

Reply:
Theorie: moet je bij een signal versturen (broadcast) druk maken of er minimaal 1 partij is die luistert. 
Parktijk: Kan me bv voorstellen dat je bepaalde hooks wilt ondersteunen als App die niet perse in elke Bundle gebruikt worden. 

In dev mode kan ik me wel voorstellen dat je wilt weten welke signals er in verdwijnen..